### PR TITLE
[AKS] Update 101-aks with latest Kubernetes versions, clarify cluster name

### PR DESCRIPTION
**Pull request by iainfoulds**
### Best Practice Checklist

- [ x ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Updated _kubernetesVersion_ to latest supported releases for 1.12, 1.11, and 1.10.
* Renamed _resourceName_ to _clusterName_ to help minimize confusion since this is specific to the cluster resource name.
  * This template will be used in a quickstart on docs.microsoft.com/azure/aks, so would like to be quite clear on this _cluster_ name parameter.

CC: @mumian @tfitzmac

### CODE REVIEW

File: 101-aks/azuredeploy.json [hunk 1]
```diff
@@ -99,7 +99,7 @@
             "apiVersion": "2018-03-31",
```
Comment (mumian) [inline 101-aks/azuredeploy.json]: There is a newer version 2019-02-01.  See https://docs.microsoft.com/en-us/azure/templates/microsoft.containerservice/2019-02-01/managedclusters

File: 101-aks/azuredeploy.json [hunk 2]
```diff
@@ -99,7 +99,7 @@
             "apiVersion": "2018-03-31",
             "type": "Microsoft.ContainerService/managedClusters",
```
Comment (mumian) [inline 101-aks/azuredeploy.json]: See the "Sort of properties" section in https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md.

File: 101-aks/azuredeploy.json (hunk 1 above)
Comment (iainfoulds) [reply 101-aks/azuredeploy.json]: There's still a couple of refinements coming there, it just went out last week. Would rather stick with 2018-03-31 at least for next couple of weeks.

Comment (mumian) [reply 101-aks/azuredeploy.json]: Make sense.

**Pull request merged**